internal/gitinterface/utils.go

```diff
@@ -37,3 +38,35 @@ func RemoteRef(refName, remoteName string) string {
 
 	return remotePath
 }
+
+// IsNiceGitVersion determines whether the version of git is "nice". Certain Git
+// subcommands that gittuf uses were added in newer versions than some common
+// client versions. Instead of using a workaround for all clients, we determine
+// if we can use the newer features or instead need to use workarounds.
+func IsNiceGitVersion() (bool, error) {
```

Comment:
Rather than enumerating capabilities, let's assess passing in the minimum version instead when the time comes. That way, the caller can control what the minimum version is where we'd need to know anyway.